src/redprl/refiner.fun

```diff
@@ -807,24 +815,18 @@ struct
          | Syn.DFUN _ => DFun.Elim
          | Syn.RECORD _ => Record.Elim
          | Syn.PATH_TY _ => Path.Elim
+         | Syn.EQUALITY _ => InternalizedEquality.Elim
          | _ => raise E.error [Fpp.text "Could not find suitable elimination rule for", TermPrinter.ppTerm ty]
 
-      fun StepEq ty =
-        case Syn.out ty of
-           Syn.BOOL => Bool.Elim
-         | _ => raise E.error [Fpp.text "Could not find suitable elimination rule for", TermPrinter.ppTerm ty]
+      val FromEq = Equality.RewriteTrue (* todo: rewrite other kinds of goals *)
```

Comment:
@favonia Yeah... I think that rewriting is not really elimination, so it makes sense to give it its own tactic.